src/View/Helper/FormHelper.php

```diff
@@ -1441,10 +1443,28 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
     {
         $context = $this->_getContext();
 
+        $options += [
+            'templateVars' => []
+        ];
+
         if (!isset($options['required']) && $options['type'] !== 'hidden') {
             $options['required'] = $context->isRequired($fieldName);
         }
 
+        if (method_exists($context, 'getRequiredMessage')) {
+            $message = $context->getRequiredMessage($fieldName);
+
+            if ($options['required'] && $message) {
+                $options['templateVars']['requiredMessage'] = $message;
```

Comment:
Perhaps use name `customValidityMessage` instead for similarity with `autoSetCustomValidity` option? Which also makes me think if `getRequiredMessage()` should be renamed.

I don't care much about specific names but I would be nice to have all methods and options related to this feature similarly named.